app_dart/lib/src/request_handlers/flush_cache.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../request_handling/exceptions.dart';
+import '../service/cache_service.dart';
+
+/// Trigger a cache flush on a config key and return empty response if successful.
+///
+/// If [cacheKeyParam] is not passed, throws [BadRequestException].
+/// If the cache does not have the given key, throws [NotFoundException].
+@immutable
+class FlushCache extends ApiRequestHandler<Body> {
+  const FlushCache(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @required this.cache,
+  }) : super(config: config, authenticationProvider: authenticationProvider);
+
+  final CacheService cache;
+
+  static const String cacheKeyParam = 'key';
+
+  @override
+  Future<Body> get() async {
+    final String cacheKey = request.uri.queryParameters[cacheKeyParam];
+    if (cacheKey == null) {
+      throw const BadRequestException('Missing required query parameter: $cacheKeyParam');
+    }
```

Comment:
Can be replaced with function `checkRequiredParameters`.

Reply:
This is a GET request and is passing via query params. It looks like that function is for verifying post data.

I added a `checkRequiredQueryParameters` there for verifying query parameters.